flink-test-utils-parent/flink-test-utils-junit/src/main/java/org/apache/flink/core/testutils/ManuallyTriggeredScheduledExecutorService.java

```diff
@@ -166,6 +166,19 @@ public boolean awaitTermination(long timeout, TimeUnit unit) {
     // Execution triggering and access to the queued tasks
     // ------------------------------------------------------------------------
 
+    /**
+     * Executes all runnable and scheduled non-periodic tasks until none are left to run. This is
+     * essentially a combination of {@link #triggerAll()} and {@link
+     * #triggerNonPeriodicScheduledTasks()} that allows making a test agnostic of how exactly a
+     * runnable is passed to the executor.
+     */
+    public void triggerAllNonPeriodicTasks() {
```

Comment:
Just to make me understand: The method is added to cover cases where a `Runnable` or non-periodic scheduled task is added while this method is already processed? Hence, it has benefits in tests where there is no explicit wait before triggering the execution of those tasks?

Reply:
This method only exists to cover both runnable and non-periodic scheduled tasks, for the simple reason that I always had to check "is the code using execute() or schedule()?" and that became rather error-prone and annoying very quickly.